src/main/java/org/cryptomator/linux/tray/AppindicatorTrayMenuController.java

```diff
@@ -0,0 +1,116 @@
+package org.cryptomator.linux.tray;
+
+import org.cryptomator.integrations.common.CheckAvailability;
+import org.cryptomator.integrations.common.OperatingSystem;
+import org.cryptomator.integrations.common.Priority;
+import org.cryptomator.integrations.tray.ActionItem;
+import org.cryptomator.integrations.tray.SeparatorItem;
+import org.cryptomator.integrations.tray.SubMenuItem;
+import org.cryptomator.integrations.tray.TrayIconLoader;
+import org.cryptomator.integrations.tray.TrayMenuController;
+import org.cryptomator.integrations.tray.TrayMenuException;
+import org.cryptomator.integrations.tray.TrayMenuItem;
+import org.purejava.appindicator.GCallback;
+import org.purejava.appindicator.MemoryAllocator;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.lang.foreign.Arena;
+import java.lang.foreign.MemorySegment;
+import java.lang.foreign.SegmentScope;
+import java.util.List;
+import java.util.function.Consumer;
+
+import static org.purejava.appindicator.app_indicator_h.*;
+
+@Priority(1000)
+@OperatingSystem(OperatingSystem.Value.LINUX)
+public class AppindicatorTrayMenuController implements TrayMenuController {
+
+	private static final Logger LOG = LoggerFactory.getLogger(AppindicatorTrayMenuController.class);
+
+	private static final SegmentScope SCOPE = SegmentScope.global();
+	private MemorySegment indicator;
+	private MemorySegment menu = gtk_menu_new();
+
+	@CheckAvailability
+	public static boolean isAvailable() {
+		return MemoryAllocator.isLoadedNativeLib();
+	}
+
+	@Override
+	public void showTrayIcon(Consumer<TrayIconLoader> iconLoader, Runnable runnable, String s) throws TrayMenuException {
+		TrayIconLoader.FreedesktopIconName callback = this::showTrayIconWithSVG;
+		iconLoader.accept(callback);
+		gtk_widget_show_all(menu);
+		app_indicator_set_menu(indicator, menu);
+		app_indicator_set_status(indicator, APP_INDICATOR_STATUS_ACTIVE());
+	}
+
+	private void showTrayIconWithSVG(String s) {
+		try (var arena = Arena.openConfined()) {
+			indicator = app_indicator_new(arena.allocateUtf8String("org.cryptomator.Cryptomator"),
+					arena.allocateUtf8String(s),
+					APP_INDICATOR_CATEGORY_APPLICATION_STATUS());
```

Comment:
I guess the `id` parameter of `app_indicator_new(...)` should be a constant (`private static final String APP_INDICATOR_ID = "org.cryptomator.Cryptomator"`). Or should the value be something like "cryptomator-status-icon"? According to the docs:

> The id parameter should be unique to your application.

I think this is only really required for referencing the icon in future calls.

```suggestion
			indicator = app_indicator_new(arena.allocateUtf8String(APP_INDICATOR_ID),
					arena.allocateUtf8String(s),
					APP_INDICATOR_CATEGORY_APPLICATION_STATUS());
```

Reply:
The ìd` is meant to identify the correct appindicator icon on D-Bus and should refer to the application, the appindicator icon belongs to. For me, the reverse URL notation is even more unique.